### PR TITLE
feat: add premium-purchase event rule [Donot Merge]

### PR DESCRIPTION
**Pull request by sri-kirsh**
## Goal
 add premium-purchase event rule. this will be consumed by transactional-email service.

Ticket: INFRA-744

### CODE REVIEW

File: .aws/src/event-rules/account-delete-monitor/index.ts [hunk 1]
```diff
@@ -58,4 +59,35 @@ export class AccountDeleteMonitorEvents extends Resource {
       userEventRuleProps
     );
   }
+
+  /**
+   * rule that attaches `user-merge` event from web repo to the
+   * account-delete-monitor lambda
+   * @private
+   */
+  private createUserMergeRules() {
```
Comment (sri-kirsh) [inline .aws/src/event-rules/account-delete-monitor/index.ts:68]: todo: carve out as seperate PR. 

this needs to go first to replay events

File: .aws/src/event-rules/account-delete-monitor/config.ts [hunk 2]
```diff
@@ -7,5 +7,13 @@ export const config = {
     schema: 'queue-check-delete',
     bus: 'default',
   },
+  userMerge: {
+    name: 'user-merge',
+    schema: 'user-merge',
+    //defined in web repo under UserMergeEvent class
+    source: 'user-merge',
+    detailType: ['web-repo'],
```
Comment (jpetto) [inline .aws/src/event-rules/account-delete-monitor/config.ts:15]: i'm a little unsure about the intention of `source` and `detailType`, but just based on the names, it seems these would be better the other way around? like `source: 'web-repo'` and `detailType: ['user-merge']`. ignore me if i am lacking context 😄

**Review by jpetto — APPROVED** [review]
code-wise this looks pretty solid to me. one question about values for one of the rules, but doesn't need to block.

File: .aws/src/event-rules/account-delete-monitor/config.ts (hunk 2 above)
Comment (sri-kirsh) [reply .aws/src/event-rules/account-delete-monitor/config.ts:15]: yes you are right..but initially i set source as `user-merge` and now i am setting it like this to replay events 🤦‍♀️ 

 so keeping it this way..after replaying events, will swap it to correct source in both web repo, ADM and this repo..carving this out as a seperate ticket here: https://getpocket.atlassian.net/browse/INFRA-770

**Comment by sri-kirsh** [comment]
closing this as its outdated and has a lot of cross over with changes already pushed..

will port the `Premium Purchase` code in a different PR

**Pull request closed without merging**